File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import Data.Char (isAscii, isLetter, toLower)
import Data.List (sort)
import qualified Data.List.NonEmpty as NE

alphabet :: String
alphabet = "abcdefghijklmnopqrstuvwxyz"

isPangram :: String -> Bool
isPangram =
  (== alphabet)
    . map NE.head
    . NE.group
    . sort
    . filter (\c -> isLetter c && isAscii c)
    . map toLower

File: zig/pangram/pangram.zig
const ascii = @import("std").ascii;
const alphabet = "abcdefghijklmnopqrstuvwxyz";

pub fn isPangram(str: []const u8) bool {
    var marks = [_]bool{false} ** 26;

    for (str) |strChar| {
        for (alphabet) |alphabetChar, i| {
            if (ascii.toLower(strChar) == alphabetChar) {
                marks[i] = true;
                break;
            }
        }
    }

    for (marks) |mark| if (mark == false) return false;
    return true;
}
